platform/android/os_android.cpp

```diff
@@ -480,12 +480,74 @@ void OS_Android::process_hover(int p_type, Point2 p_pos) {
 	}
 }
 
-void OS_Android::process_double_tap(Point2 p_pos) {
+void OS_Android::process_mouse_pressed(int p_button, Point2 p_pos, bool pressed) {
+	// https://developer.android.com/reference/android/view/MotionEvent.html#ACTION_HOVER_ENTER
 	Ref<InputEventMouseButton> ev;
 	ev.instance();
 	ev->set_position(p_pos);
 	ev->set_global_position(p_pos);
-	ev->set_pressed(false);
+	ev->set_pressed(pressed);
+	switch (p_button) {
+		case 1: //left button
+			ev->set_button_index(BUTTON_LEFT);
+			break;
+		case 2: //right button
+			ev->set_button_index(BUTTON_RIGHT);
+			break;
+		case 4: //middle button
+			ev->set_button_index(BUTTON_MIDDLE); //pc middle mouse button
+			break;
+		case 5: // wheel down
+			ev->set_button_index(BUTTON_WHEEL_DOWN);
+			ev->set_factor(1);
+			break;
+		case 6: // wheel up
+			ev->set_button_index(BUTTON_WHEEL_UP);
+			ev->set_factor(1);
+			break;
+	}
+	input->parse_input_event(ev);
+}
+
+void OS_Android::process_mouse_moved_pressed(int p_button, Point2 p_pos) {
+	// https://developer.android.com/reference/android/view/MotionEvent.html#ACTION_HOVER_ENTER
+	Ref<InputEventMouseMotion> ev;
+	ev.instance();
+	ev->set_position(p_pos);
+	ev->set_global_position(p_pos);
+	ev->set_relative(p_pos - hover_prev_pos);
+	switch (p_button) {
+		case 1: //left button
+			ev->set_button_mask(BUTTON_MASK_LEFT);
+			break;
+		case 2: //right button
+			ev->set_button_mask(BUTTON_MASK_RIGHT);
+			break;
+		case 4: //middle button
+			ev->set_button_mask(BUTTON_MASK_MIDDLE);
+			break;
+	}
+	input->parse_input_event(ev);
+	hover_prev_pos = p_pos;
+}
+
+void OS_Android::process_double_tap(Point2 p_pos, int p_button) {
+	Ref<InputEventMouseButton> ev;
+	ev.instance();
+	ev->set_position(p_pos);
+	ev->set_global_position(p_pos);
+	ev->set_pressed(p_button != 0);
```

Comment:
Why is the `pressed` state dependent on the button id?

Reply:
because doubletap by touch currently does not add pressed so I thought it is intentional, and I don't want to brick anything, so added pressed only if it's fired by mouse

Reply:
also any button that sends pressed true need to also send false in another event, so in case of mouse doubletap happns before `onGenericMotionEvent` so release is sent automaticlly after